tests/src/Traits/OgTrait.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Drupal\joinup\Traits;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\og\Entity\OgMembership;
+use Drupal\og\Og;
+use Drupal\og\OgMembershipInterface;
+
+/**
+ * Contains helper methods regarding the organic groups.
+ *
+ * @package src\Traits
+ */
+trait OgTrait {
+
+  /**
+   * Checks that with a given og role is already logged in.
+   *
+   * If the user has more than the required roles, he might have permissions
+   * from the rest of the roles that will lead the test to a false positive.
+   * For this reason, we request check for the specific roles passed.
+   *
+   * @param array $roles
+   *   An array of roles to check.
+   * @param \Drupal\Core\Entity\EntityInterface $group
+   *   The group that is checked if the user has the role.
+   *
+   * @return bool
+   *   Returns TRUE if the current logged in user has this role (or roles).
+   */
+  protected function loggedInWithOgRole($roles, EntityInterface $group) {
+    if (!$this->loggedIn() || !$this->user) {
+      return FALSE;
+    }
+    $user = \Drupal::entityTypeManager()->getStorage('user')->loadUnchanged($this->user->uid);
+    $membership = Og::getMembership($user, $group);
+    if (empty($membership)) {
+      return FALSE;
+    }
+    if ($roles == $membership->getRolesIds()) {
+      return FALSE;
+    }
```

Comment:
Also, why does this return FALSE instead of throwing an exception when the assertion fails? That means that on the calling end we will always have to check the return value, and throw an exception there.